cfn/guardduty.template

```diff
@@ -338,7 +379,7 @@
             "DependsOn": [
                 "LambdaKmsKey",
                 "EncryptLambdaFunction",
-                "EncryptLambdaPolicy"
+                "BasicLambdaPolicy"
```

Comment:
@kkuzmin we need to make sure that `EndpointAPI` CustomResource is called **after**  BasicLambdaPolicy is in place (as otherwise it will be run, but it's possible to have 0 logs there)

could you please:

* add `BasicLambdaPolicy` as dependency to `EndpointAPI` resource:
https://github.com/alertlogic/cwe-collector/blob/37d2923a187a47f30a5633e5ac2c880e211ff02c/cfn/guardduty.template#L521 
* either remove dependency `CollectLambdaRole` (as we already have Policy as deps), or rename it  to `BasicLambdaRole`. If latter, please do the same for `EncryptSecretKeyCustomResource` by adding  deps to `BasicLambdaRole` here as well. (Let's stay consistent, either we always mention Role+Policy, or policy only). I'd go for both - role & policy as it's more explicit way.

Reply:
`BasicLambdaPolicy` can't be added as a dependency to `EndpointAPIs` because there is a circle in this case:
```
EncryptLambdaFunction -> EndpointAPIs -> BasicLambdaPolicy -> EncryptLambdaFunction
```